src/models/FilterStakeholder.js

```diff
@@ -0,0 +1,148 @@
+// import { stringToSearchRegExp } from '../utilities';
+
+/*
+ * A simple model for grower filter
+ */
+
+export const ALL_ORGANIZATIONS = 'ALL_ORGANIZATIONS';
+export const ORGANIZATION_NOT_SET = 'ORGANIZATION_NOT_SET';
+
+export default class Filter {
+  constructor(options) {
+    Object.assign(this, options);
+  }
+
+  getWhereObj() {
+    let where = {};
+
+    if (this.type) {
+      where.type = this.type;
+    }
+
+    if (this.name) {
+      const [firstName, ...lastName] = this.name.split(' ');
+      if (this.type === 'Person') {
+        if (this.firstName) {
+          where.firstName = firstName;
+        }
+
+        if (this.lastName) {
+          where.lastName = lastName[0];
+        }
+        where.orgName = '';
+      } else {
+        // give all name fields the value for filter
+        where.orgName = this.name;
+        where.firstName = firstName || this.name;
```

Comment:
Setting the `firstName` and `lastName` in the filter for a non-person doesn't seem right to me.

Reply:
I wonder if the logic for constructing a name filter is the sort of more sophisticated task we should delegate to the API.